src/app/chip/chip-input/chip-input.component.ts

```diff
@@ -0,0 +1,95 @@
+import { Component, ElementRef, EventEmitter, forwardRef, Input, NgZone, OnDestroy, OnInit, Output } from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+
+@Component({
+  selector: 'mz-chip-input',
+  templateUrl: './chip-input.component.html',
+  styleUrls: ['./chip-input.component.scss'],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => MzChipInputComponent),
+      multi: true,
+    },
+  ],
+})
+export class MzChipInputComponent implements ControlValueAccessor, OnInit, OnDestroy {
+  @Input() autocompleteOptions: Materialize.AutoCompleteOptions;
+  @Input() placeholder: string;
+  @Input() secondaryPlaceholder: string;
+  @Output() add = new EventEmitter<Materialize.ChipDataObject>();
+  @Output() delete = new EventEmitter<Materialize.ChipDataObject>();
+  @Output() select = new EventEmitter<Materialize.ChipDataObject>();
+
+  get value(): Materialize.ChipDataObject[] {
+    return this.chipInputElement.material_chip('data') as Materialize.ChipDataObject[];
+  }
+
+  private chipInputElement: JQuery;
+  private onChangeCallback = (_: Materialize.ChipDataObject[]) => {};
+
+  constructor(
+    private elementRef: ElementRef,
+    private zone: NgZone,
+  ) { }
+
+  ngOnInit() {
+    this.initElements();
+    this.initMaterializeChip();
+  }
+
+  ngOnDestroy() {
+    this.chipInputElement.off('chip.add');
+    this.chipInputElement.off('chip.delete');
+    this.chipInputElement.off('chip.select');
+  }
+
+  initElements() {
+    this.chipInputElement = $(this.elementRef.nativeElement);
+  }
+
+  initMaterializeChip(value?: Materialize.ChipDataObject[]) {
+    // fix issue autocomplete is not a function
+    // https://github.com/Dogfalo/materialize/issues/4401
+    this.zone.runOutsideAngular(() => {
+      setTimeout(() => {
```

Comment:
Do you know if without the setTimeOut the chip component still work?

Reply:
It gives an error, see the issue link in comment below the code